subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/SourceSetUtil.java

```diff
@@ -47,7 +48,7 @@ private static void configureForSourceSet(final SourceSet sourceSet, final Sourc
             public Object call() {
                 return sourceSet.getCompileClasspath().plus(target.files(sourceSet.getJava().getOutputDir()));
             }
-        });
+        }).cache();
```

Comment:
Did you ask if this method here is safe to use? I am not 100% sure, yet...

Reply:
No... after reading the code, I think it's not safe as it might not be able to update new values. I switch to more verbose `ConfigurableFileCollection`.